### ISSUE TITLE
[LSP] `"settings.configurationPreference: "filesystemFirst"` doesn't work as expected

### ISSUE THREAD

**Issue by rchl** [astral-sh/ruff#24420]
### Summary

This is an issue specific to LSP since it relies on `settings.configurationPreference` being overridden.

ruff lsp seems to not to be proffering local filesystem configuration when `filesystemFirst` value is used.

1. Clone this very simple project that includes isort rules and enables linting of imports: https://github.com/rchl/test-ruff
2. Open this project in your editor and note that imports are sorted according to project rules and no diagnostics show up for organizing imports - that is, the specified rules are followed.
3. Create a global configuration in `~/ruff-cfg.toml` for example (on purpose not using location that ruff would pick up by itself):
  ```toml
  line-length = 120
  
  [lint.isort]
  case-sensitive = false
  force-single-line = true
  from-first = true
  no-sections = true
  order-by-type = false
  required-imports = ["from __future__ import annotations"]
  ```
4. Set editor settings (I use Sublime Text but I'll provide example for VSCode since that it's more widely used):
  ```json
      "ruff.configurationPreference": "filesystemFirst",
      "ruff.configuration": "~/ruff-cfg.toml",
  ```

Expected:
- Project-local settings should be used - ruff should not raise issue about imports being unsorted.
- Triggering "organize imports" (if available) should not reformat imports.

What happens:
- The imports are considered not sorted.
- The configuration in the home directory is either prioritized or merged with project-local settings.

### Version

ruff 0.15.9

**Comment by MichaReiser** [comment]
Hi

I can see how the `filesystemFirst` name can be confusing when setting a path in `configuration`. 

A better name for `filesystemFirst` might have been project configuration first, which is also closer to its description:

> Configuration files present in the workspace takes priority over editor settings.

What you want is `editorFirst`, you want that the `configuration` value from your editor configuration takes precedence over any project specific configuration.

**Comment by rchl** [comment]
Sorry but this makes no sense to me. I wonder if your understanding of my issue is backwards.

I want the configuration from `pyproject.toml` in my `test-ruff` project to be effectively used even though I have set

```json
    "ruff.configurationPreference": "filesystemFirst",
    "ruff.configuration": "~/ruff-cfg.toml",
```

in the LSP settings in my editor (meant to be used as a fallback when no file-based configuration is present).

What happens is the opposite - the LSP configuration is effectively used. Using `editorFirst` doesn't help with that goal.

**Comment by MichaReiser** [comment]
Yes sorry, I misunderstood you. 

I think what you want is `filesystemOnly`, which does not exist (see https://github.com/astral-sh/ruff/issues/16209). `filesystemFirst` loads the editor configuration (`~/ruff-cfg.toml`), and then overrides the settings that are also specified in your `pyproject.toml`. Meaning, the configuration in the project's `pyproject.toml` take precedence over what's in `ruff-cfg.toml`. However, settings that are only in `ruff-cfg.toml` are not overridden.

For now, you could use VS Code's workspace configuration to only set `ruff.configuration` in the projects where you want the shared configuration or unset `ruff.configuration` in projects where you don't want the shared configuration to apply.

**Comment by rchl** [comment]
> I think what you want is `filesystemOnly`, which does not exist (see [#16209](https://github.com/astral-sh/ruff/issues/16209))

Technically I would still want to fallback to editor configuration if project does not have configuration so not sure if `filesystemOnly` would be a good name.

> For now, you could use VS Code's workspace configuration to only set `ruff.configuration` in the projects where you want the shared configuration or unset `ruff.configuration` in projects where you don't want the shared configuration to apply.

Unfortunately my setup consists of a project with about 30 folders it in, each might or might not include its own ruff config. So setting `ruff.configuration` in workspace configuration won't help.

**Comment by rchl** [comment]
Will close this since linked issue does seem to discuss this specific issue also.
